src/es_archiver/lib/indices/delete_index.ts

```diff
@@ -18,22 +18,34 @@
  */
 
 import { get } from 'lodash';
+import { Client } from 'elasticsearch';
+import { ToolingLog } from '@kbn/dev-utils';
+import { Stats } from '../stats';
 
 // see https://github.com/elastic/elasticsearch/blob/99f88f15c5febbca2d13b5b5fda27b844153bf1a/server/src/main/java/org/elasticsearch/cluster/SnapshotsInProgress.java#L313-L319
 const PENDING_SNAPSHOT_STATUSES = ['INIT', 'STARTED', 'WAITING'];
 
-export async function deleteIndex(options) {
+export async function deleteIndex(options: {
+  client: Client;
+  stats: Stats;
+  index: string;
+  log: ToolingLog;
+  retryIfSnapshottingCount?: number;
+}): Promise<void> {
   const { client, stats, index, log, retryIfSnapshottingCount = 10 } = options;
 
   const getIndicesToDelete = async () => {
     const aliasInfo = await client.indices.getAlias({ name: index, ignore: [404] });
-    return aliasInfo.status === 404 ? index : Object.keys(aliasInfo);
+    return aliasInfo.status === 404 ? [index] : Object.keys(aliasInfo);
   };
 
   try {
     const indicesToDelete = await getIndicesToDelete();
     await client.indices.delete({ index: indicesToDelete });
-    stats.deletedIndex(indicesToDelete);
+    for (let i = 0; i < indicesToDelete.length; i++) {
+      const indexToDelete = indicesToDelete[i];
+      stats.deletedIndex(indexToDelete);
+    }
```

Comment:
In case of aliases, we were calling `stats.deletedIndex` with the list of indices, which resulted on incorrect behavior (the call expects a string). Fixed it with a loop.